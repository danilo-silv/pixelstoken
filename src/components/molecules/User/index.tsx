import { LinearGradient } from 'expo-linear-gradient';
import { XStack, YStack, Avatar, Text, H4 } from 'tamagui';

export function User() {
  return (
    <XStack space="$2" ai="center">
      <LinearGradient
        colors={['#0AFF96', '#00E0FF']}
        start={[0, 0]}
        end={[1, 1]}
        style={{
          padding: 3,
          borderRadius: 50,
        }}>
        <Avatar size="$5" circular>
          <Avatar.Image src="https://ca.slack-edge.com/THQS6JHC1-U072Q9RBV1T-f48dbe97c1e5-512" />
          <Avatar.Fallback backgroundColor="$gray5" />
        </Avatar>
      </LinearGradient>
      <YStack>
        <Text color="white">Ol√°,</Text>

        <H4 color="white" fontWeight="bold" mt="$-2">
          Danilo
        </H4>
      </YStack>
    </XStack>
  );
}
