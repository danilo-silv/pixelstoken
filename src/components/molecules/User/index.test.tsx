import { render } from '@test-utils';

import { User } from '../User';

describe('User Component', () => {
  describe('Render', () => {
    it('should match snapshot', () => {
      const { toJSON } = render(<User />);
      expect(toJSON()).toMatchSnapshot();
    });

    describe('Texts', () => {
      it('should render the greeting text', () => {
        const { getByText } = render(<User />);
        expect(getByText('Hello,')).toBeTruthy();
      });

      it('should render the username text', () => {
        const { getByText } = render(<User />);
        expect(getByText('Danilo')).toBeTruthy();
      });
    });

    describe('Avatar', () => {
      it('should render the avatar image with correct src', () => {
        const { getByTestId } = render(<User />);
        const avatarImage = getByTestId('avatar-image');
        expect(avatarImage.props.source.uri).toBe(
          'https://ca.slack-edge.com/THQS6JHC1-U072Q9RBV1T-f48dbe97c1e5-512'
        );
      });
    });

    describe('Linear Gradient', () => {
      it('should render the gradient with correct colors', () => {
        const { getByTestId } = render(<User />);
        const gradient = getByTestId('gradient');

        expect(gradient.props.colors).toEqual([4278910870, 4278247679]);
      });

      it('should set the gradient start position correctly', () => {
        const { getByTestId } = render(<User />);
        const gradient = getByTestId('gradient');
        expect(gradient.props.startPoint).toEqual([0, 0]);
      });

      it('should set the gradient end position correctly', () => {
        const { getByTestId } = render(<User />);
        const gradient = getByTestId('gradient');
        expect(gradient.props.endPoint).toEqual([1, 1]);
      });

      it('should apply the correct linear gradient style', () => {
        const { getByTestId } = render(<User />);
        const gradient = getByTestId('gradient');
        expect(gradient.props.style).toEqual({
          padding: 3,
          borderRadius: 50,
        });
      });
    });
  });
});
