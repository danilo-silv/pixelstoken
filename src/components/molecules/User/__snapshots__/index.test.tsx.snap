// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component Render should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 7,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4278910870,
        4278247679,
      ]
    }
    endPoint={
      [
        1,
        1,
      ]
    }
    startPoint={
      [
        0,
        0,
      ]
    }
    style={
      {
        "borderRadius": 50,
        "padding": 3,
      }
    }
    testID="gradient"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomLeftRadius": 100000,
          "borderBottomRightRadius": 100000,
          "borderTopLeftRadius": 100000,
          "borderTopRightRadius": 100000,
          "flexDirection": "column",
          "height": 52,
          "justifyContent": "center",
          "maxHeight": 52,
          "maxWidth": 52,
          "minHeight": 52,
          "minWidth": 52,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "relative",
          "width": 52,
        }
      }
    >
      <View
        style={
          {
            "bottom": 0,
            "flexDirection": "column",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      >
        <Image
          fullscreen={true}
          onError={[Function]}
          onLoad={[Function]}
          source={
            {
              "uri": "https://ca.slack-edge.com/THQS6JHC1-U072Q9RBV1T-f48dbe97c1e5-512",
            }
          }
          style={
            {
              "height": 52,
              "width": 52,
            }
          }
          testID="avatar-image"
        />
      </View>
      <View
        style={
          {
            "bottom": 0,
            "flexDirection": "column",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
  <View
    style={
      {
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "Inter",
        }
      }
      suppressHighlighting={true}
    >
      Hello,
    </Text>
    <Text
      accessibilityRole="header"
      style={
        {
          "color": "white",
          "fontFamily": "Inter",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 0,
          "lineHeight": 30,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": -7,
          "userSelect": "auto",
        }
      }
      suppressHighlighting={true}
    >
      Danilo
    </Text>
  </View>
</View>
`;
